### DISCUSSION TITLE
Aws::InitAPI(options) fails for release version dlls on Windows

### DISCUSSION THREAD

**Discussion by RoyZhang2022** [aws/aws-sdk-cpp#2762] [Q&A] [answered]
Hi,

We have some unit tests that uses AWS SDK C++ dlls on Windows. The code is like this. We build only components "core;rds;secretsmanager".

```
    static Aws::SDKOptions options;

    static void SetUpTestSuite() {
        Aws::InitAPI(options);
        ...
    }

```

I found the tests could work well with SDK dlls debug version when I build my tests as debug version. If I run my debug version test with SDK dlls release version, I always see an exception from `Aws::InitAPI(options)`. The call stack is like this. I suppose my tests should run successfully no matter I am using release version or debug version SDK dlls. But now I always see this problem. Seems `monitoringFactoryCreateFunctions` is always a random value for SDK release version. Tried 1.11.21 and 1.11.190. Both have same problem. Is this a bug or something I need to pay attention?

![image](https://github.com/aws/aws-sdk-cpp/assets/104782736/7b3c74a2-189b-4e37-a10e-6703a0f66cc7)
![image](https://github.com/aws/aws-sdk-cpp/assets/104782736/39dbde4a-5a9a-40a3-8783-cb9d92c367f2)
![image](https://github.com/aws/aws-sdk-cpp/assets/104782736/d5ff1adc-cd6c-4dd2-bed6-64a0058cce02)
![image](https://github.com/aws/aws-sdk-cpp/assets/104782736/cd8c8bcc-fa4e-4e4c-8e1d-aa3ec00eb10b)

My Windows info:
```
Edition	Windows 10 Pro
Version	22H2
Installed on	‎2022-‎05-‎03
OS build	19045.3636
Experience	Windows Feature Experience Pack 1000.19053.1000.0
```

Thanks,
Roy

**Comment by RoyZhang2022** [comment]
For debug version I found `options.monitoringOptions.customizedMonitoringFactory_create_fn` is 0.

**Comment by RoyZhang2022** [comment]
Just found the `options.monitoringOptions.customizedMonitoringFactory_create_fn` value is changed once Aws::InitAPI() is entered. 

--options value at caller of Aws::InitAPI()
![image](https://github.com/aws/aws-sdk-cpp/assets/104782736/fd74d0f8-fb3a-4aa6-a899-9ec59885c0ea)

--options value once enter Aws::InitAPI()
![image](https://github.com/aws/aws-sdk-cpp/assets/104782736/9c7b738d-3b27-4c2c-a3da-a11796d5e38a)

Even I set options.monitoringOptions value manually it still has this error.
```
        Aws::MonitoringOptions monitoringOptions;
        options.monitoringOptions = monitoringOptions;
        Aws::InitAPI(options);
```

**Comment by jmklix** [comment, marked as answer]
Here are a few things that might fix the errors that you're seeing:
- Updating to the latest version of this sdk. This [PR](https://github.com/aws/aws-sdk-cpp/pull/3197) might have fixed the errors that you were seeing
- Check that the built and installed SDK version matches the version of the SDK headers
- adding `{}` when initializing options (i.e. `static Aws::SDKOptions options = {};`) 

Please let us know if you are still running into errors when using this sdk
